c7n/resources/redshift.py

```diff
@@ -77,6 +76,114 @@ def __call__(self, redshift):
                 self.match(redshift.get('VpcId')) or False)
 
 
+@Redshift.filter_registry.register('logging')
+class LoggingFilter(ValueFilter):
+    """ Checks Redshift logging status and attributes.
+
+    :example:
+
+    .. code-block:: yaml
+
+
+            policies:
+
+                - name: redshift-logging-bucket-and-prefix-test
+                  resource: redshift
+                  filters:
+                   - type: logging
+                     key: LoggingEnabled
+                     value: true
+                   - type: logging
+                     key: S3KeyPrefix
+                     value: "accounts/{account_id}"
+                   - type: logging
+                     key: BucketName
+                     value: "redshiftlogs"
+
+
+    """
+    permissions = ("redshift:DescribeLoggingStatus",)
+    schema = type_schema('logging', rinherit=ValueFilter.schema)
+    annotation_key = 'c7n:logging'
+
+    def process(self, clusters, event=None):
+        client = local_session(self.manager.session_factory).client('redshift')
+        results = []
+        for cluster in clusters:
+            if self.annotation_key not in cluster:
+                try:
+                    result = client.describe_logging_status(
+                        ClusterIdentifier=cluster['ClusterIdentifier'])
+                    result.pop('ResponseMetadata')
+                except client.exceptions.ClusterNotFound:
+                    continue
+                cluster[self.annotation_key] = result
+
+            if self.match(cluster[self.annotation_key]):
+                results.append(cluster)
+        return results
+
+
+@Redshift.action_registry.register('set-logging')
+class SetRedshiftLogging(BaseAction):
+    """Action to enable/disable Redshift logging for a Redshift Cluster.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: redshift-test
+                resource: redshift
+                filters:
+                  - type: logging
+                    key: LoggingEnabled
+                    value: false
+                actions:
+                  - type: set-logging
+                    bucket: redshiftlogtest
+                    prefix: redshiftlogs
+                    state: enabled
+    """
+    schema = type_schema(
+        'set-logging',
+        state={'enum': ['enabled', 'disabled']},
+        bucket={'type': 'string'},
+        prefix={'type': 'string'},
+        required=('state',))
+
+    permissions = ("redshift:EnableLogging", "redshift:DisableLogging",
```

Comment:
fwiw, getting permissions has two forms, the simple default is just a tuple on as a class attribute, for cases like this where its variable based on policy configuration, its also possible to implement a get_permissions method that returns something differential based on configuration.